qiskit_nature/second_q/operators/electronic_integrals.py

```diff
@@ -435,6 +435,104 @@ def apply(
+        to the :attr:`alpha`, :attr:`beta`, and :attr:`beta_alpha` attributes of the provided
+        ``ElectronicIntegrals`` operands.
+
+        This method is special, because it handles the scenario in which any operand has a non-empty
+        :attr:`beta` attribute, in which case the empty-beta attributes of any other operands will
+        be filled with :attr:`alpha` attributes of those operands.
+        The same applies to the :attr:`beta_alpha` attributes.
+
+        Args:
+            function: the stacking function to apply to the internal arrays of the provided
+                operands. This function must take a sequence of numpy (or sparse) arrays as its
+                first argument. You should use :code:`functools.partial` if you need to provide
+                keyword arguments (e.g. :code:`partial(np.stack, axis=-1)`)
+            operands: a sequence of ``ElectronicIntegrals`` instances on which to operate.
+            validate: when set to False, no validation will be performed.
+
+        Returns:
+            A new ``ElectronicIntegrals``.
+        """
+        alpha = PolynomialTensor.stack(function, [op.alpha for op in operands], validate=validate)
+
+        beta: PolynomialTensor = None
+        if any(not op.beta.is_empty() for op in operands):
+            # If any beta-entry is non-empty, we have to perform this computation.
+            # Empty tensors will be populated with their alpha-terms automatically.
+            beta = PolynomialTensor.stack(
+                function,
+                [op.alpha if op.beta.is_empty() else op.beta for op in operands],
+                validate=validate,
+            )
+
+        beta_alpha: PolynomialTensor = None
+        if all(not op.beta_alpha.is_empty() for op in operands):
+            # We can only perform this operation, when all beta_alpha tensors are non-empty.
+            beta_alpha = PolynomialTensor.stack(
+                function, [op.beta_alpha for op in operands], validate=validate
+            )
+        return cls(alpha, beta, beta_alpha, validate=validate)
+
+    def split(
+        self,
+        function: Callable[..., np.ndarray | SparseArray | Number],
+        indices_or_sections: int | Sequence[int],
+        *,
+        validate: bool = True,
+    ) -> list[ElectronicIntegrals]:
+        """Exposes the :meth:`qiskit_nature.second_q.operators.PolynomialTensor.split` method.
+
+        This behaves identical to the ``split`` implementation of the ``PolynomialTensor``, applied
+        to the :attr:`alpha`, :attr:`beta`, and :attr:`beta_alpha` attributes of the provided
+        ``ElectronicIntegrals`` operands.
+
+        This method is special, because it handles the scenario in which any operand has a non-empty
+        :attr:`beta` attribute, in which case the empty-beta attributes of any other operands will
+        be filled with :attr:`alpha` attributes of those operands.
+        The same applies to the :attr:`beta_alpha` attributes.
```

Comment:
Perhaps some unit tests should be added for this method. I know that this method heavily relies on the same one in the PolynomicTensor (which is tested), but still it would be good to test it. Especially due to this special behavior

Reply:
Yeh I was kind of in a hurry. I will see if I can add some unittests similar to how it is done in the `einsum` routine.